ebuild-writing/variables/text.xml

```diff
@@ -588,5 +588,54 @@ when joining paths. For example: <c>${D%/}/</c>, <c>${ED%/}/</c>,
 </body>
 </section>
 
+<section>
+<title>Use of constant-value variables in ebuilds</title>
+<body>
+
+<p>
+Variables frequently have significant value in ebuilds, making it
+possible to avoid unnecessary repetitions and make maintenance easier.
+However, references to constant-value variables should be used with care
+as their proactive use can harm readability and increase maintenance
+burden (e.g. when renaming the package). In particular, using variables
+whose values are not strictly guaranteed to match the expected string
+should be avoided.
+</p>
+
+<p>The examples of beneficial constant-value variable references are:</p>
+<ul>
+  <li>
+    using <c>${PV}</c> and the variables derived from it (<c>${P}</c>,
+    <c>${MY_P}</c>…) in <c>SRC_URI</c> and <c>S</c> to avoid having
+    to update those variables for every version bump (presuming that
+    <c>${PV}</c> is used to indicate the upstream version);
+  </li>
+
+  <li>
+    using <c>${PF}</c> in <c>--docdir</c> path — this is a canonical
+    Gentoo path that is always required to match <c>${PF}</c>.
+  </li>
+</ul>
+
+<p>The examples of bad constant-value variable references are:</p>
+<ul>
+  <li>
+    using <c>${PN}</c> in <c>HOMEPAGE</c>, <c>EGIT_REPO_URI</c>,
+    the domain or path prefix part of <c>SRC_URI</c> — it breaks URL
+    parsing in editors and terminals, and makes it hard to copy the link
+    for external use (e.g. while reviewing via gitweb),
```

Comment:
That presumes that people agree to it. I agree that it's incorrect to use `${PN}` when you actually mean the literal string that it evaluates to. The argument is that if you rename a package from `foo` to `bar` with the upstream staying constant, your `${PN}` in SRC_URI will break things. In that case, however, `${P}` will also change. So if you had `${P}.tar.gz` in SRC_URI, that will also break. If you had `-> ${P}.tar.gz`, that will require a re-run of `repoman manifest`.

The sole argument that it makes copy-paste harder mostly applies to users submitting pull requests as opposed to developers pushing their ebuilds directly to the repo. That, in turn, can be enforced by proxy-maint directly.